File: app/src/main/java/com/aaronbrecher/neverlate/adapters/SkuListAdapter.java
package com.aaronbrecher.neverlate.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.aaronbrecher.neverlate.databinding.SkuListItemBinding;
import com.aaronbrecher.neverlate.interfaces.ListItemClickListener;
import com.android.billingclient.api.SkuDetails;

import java.util.List;

public class SkuListAdapter extends RecyclerView.Adapter<SkuListAdapter.SkuViewHolder> {

    private List<SkuDetails> mSkuList;
    private ListItemClickListener mClickListener;

    public SkuListAdapter(List<SkuDetails> skuList, ListItemClickListener clickListener) {
        mSkuList = skuList;
        mClickListener = clickListener;
    }

    @NonNull
    @Override
    public SkuViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
        return new SkuViewHolder(SkuListItemBinding.inflate(inflater, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull SkuViewHolder skuViewHolder, int position) {
        SkuDetails details = mSkuList.get(position);
        skuViewHolder.mBinding.setSkuDetails(details);
        String title = details.getTitle();
        if(title.contains("(")){
            title = title.substring(0, title.indexOf("("));
        }

        skuViewHolder.mBinding.skuListItemName.setText(title);
    }

    @Override
    public int getItemCount() {
        return mSkuList == null ? 0 : mSkuList.size();
    }

    class SkuViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        SkuListItemBinding mBinding;
        public SkuViewHolder(SkuListItemBinding binding) {
            super(binding.getRoot());
            mBinding = binding;
            mBinding.skuListItemPurchaseButton.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            SkuDetails details = mSkuList.get(getAdapterPosition());
            mClickListener.onListItemClick(details);
        }
    }

    public void swapLists(List<SkuDetails> skuList){
        this.mSkuList = skuList;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/aaronbrecher/neverlate/adapters/SkuListAdapter.kt
package com.aaronbrecher.neverlate.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.aaronbrecher.neverlate.databinding.SkuListItemBinding
import com.aaronbrecher.neverlate.interfaces.ListItemClickListener
import com.android.billingclient.api.SkuDetails

import androidx.annotation.NonNull
import androidx.recyclerview.widget.RecyclerView

class SkuListAdapter(private var mSkuList: List<SkuDetails>?, private val mClickListener: ListItemClickListener) : RecyclerView.Adapter<SkuListAdapter.SkuViewHolder>() {

    @NonNull
    override fun onCreateViewHolder(@NonNull viewGroup: ViewGroup, viewType: Int): SkuViewHolder {
        val inflater = LayoutInflater.from(viewGroup.context)
        return SkuViewHolder(SkuListItemBinding.inflate(inflater, viewGroup, false))
    }

    override fun onBindViewHolder(@NonNull skuViewHolder: SkuViewHolder, position: Int) {
        val details = mSkuList!![position]
        skuViewHolder.mBinding.skuDetails = details
        var title = details.title
        if (title.contains("(")) {
            title = title.substring(0, title.indexOf("("))
        }

        skuViewHolder.mBinding.skuListItemName.text = title
    }

    override fun getItemCount(): Int {
        return if (mSkuList == null) 0 else mSkuList!!.size
    }

    inner class SkuViewHolder(var mBinding: SkuListItemBinding) : RecyclerView.ViewHolder(mBinding.root), View.OnClickListener {
        init {
            mBinding.skuListItemPurchaseButton.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            val details = mSkuList!![adapterPosition]
            mClickListener.onListItemClick(details)
        }
    }

    fun swapLists(skuList: List<SkuDetails>) {
        this.mSkuList = skuList
        notifyDataSetChanged()
    }
}
